torch/profiler/profiler.py

```diff
@@ -3,12 +3,185 @@
 import os
 import tempfile
 from enum import Enum
-from typing import Any, Callable, Iterable, Optional
+from functools import partial
+from typing import Any, Callable, Dict, Iterable, List, Optional, Tuple
 from warnings import warn
 
 import torch
 import torch.autograd.profiler as prof
-from torch.autograd import kineto_available, ProfilerActivity
+from torch.autograd import ProfilerActivity, kineto_available
+
+
+def supported_activities():
+    """
+    Returns a set of supported profiler tracing activities.
+
+    Note: profiler uses CUPTI library to trace on-device CUDA kernels.
+    In case when CUDA is enabled but CUPTI is not available, passing
+    ``ProfilerActivity.CUDA`` to profiler results in using the legacy CUDA
+    profiling code (same as in the legacy ``torch.autograd.profiler``).
+    This, in turn, results in including CUDA time in the profiler table output,
+    but not in the JSON trace.
+    """
+    return torch.autograd._supported_activities()
+
+
+class kineto_profile(object):
```

Comment:
Can we name this `_KinetoProfile`? I don't want to commit to expanding the public API until we (a) know that we've drawn the boundaries where we like and (b) have killed the autograd profiler. It's probably fine to leave an entry in the API Reference as long as it is marked experimental.

Reply:
How to make it as experimental in API Reference? are there any guidance?

Reply:
It seems that we only need add a warning in the profiler.rst file?

Reply:
Just prefixing with an underscore should be sufficient.